standalone_app/src/sqlite3.ts

```diff
@@ -38,32 +38,47 @@ export const loadStorage = (
     )
     db.checkRc(rc)
     try {
-      if (!isSupportedSchema(db)) {
+      const schemaVersion = getSchemaVersion(db)
+      if (!isSupportedSchema(schemaVersion)) {
         return
       }
-      const studies = getStudies(db)
+      const studies = getStudies(db, schemaVersion)
       setter((prev) => [...prev, ...studies])
     } finally {
       db.close()
     }
   })
 }
 
-const isSupportedSchema = (db: SQLite3DB): boolean => {
-  let supported = true
+const getSchemaVersion = (db: SQLite3DB): string => {
+  let schemaVersion = ""
   db.exec({
-    sql: "SELECT schema_version FROM version_info LIMIT 1",
+    sql: "SELECT version_num FROM alembic_version LIMIT 1",
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     callback: (vals: any[]) => {
-      if (vals[0] != 12) {
-        supported = false
-      }
+      schemaVersion = vals[0]
     },
   })
-  return supported
+  return schemaVersion
+}
+
+const isSupportedSchema = (schemaVersion: string): boolean => {
+  const lowestVersion = "v2.6.0.a" // supported: "v3.2.0.a", "v3.0.0.{a,b,c,d}", "v2.6.0.a"
+  if (schemaVersion == lowestVersion) return true
+  return isGreaterSchemaVersion(schemaVersion, lowestVersion)
+}
+
+const isGreaterSchemaVersion = (
+  leftVersion: string,
+  rightVersion: string
+): boolean => {
+  // return leftVersion > rightVersion
+  leftVersion = leftVersion.replace(/\D/g, "")
+  rightVersion = rightVersion.replace(/\D/g, "")
+  return Number(leftVersion) > Number(rightVersion)
 }
```

Comment:
Can we simply write this logic like below?

```suggestion
const isSupportedSchema = (schemaVersion: string): boolean => {
  return ["v3.2.0", "v3.0.0", "v2.6.0"].some((v) => schemaVersion.startsWith(v))
}
```

Reply:
The `isGreaterSchemaVersion` function is used for switching to queries described later, so I think the number of lines will not significantly decrease further.